pkg/download/classic/download.go

```diff
@@ -261,7 +243,7 @@ func shouldFilter() bool {
 	return featureflags.DownloadFilter().Enabled() && featureflags.DownloadFilterClassicConfigs().Enabled()
 }
 
-func downloadAndUnmarshalConfig(client dtclient.Client, theApi api.API, value value) ([]map[string]interface{}, error) {
+func download(client dtclient.Client, theApi api.API, value value) (map[string]any, error) {
```

Comment:
You didn't introduce it, but the logic of this function would benefit from some explaining comments. That fact that it does a "read config by ID" call to list all configs of either a top-level API (id = "" so we just do a <api>/ call) or list all children of a parent API (<api>/<parentid>/<subpath>) confused me when trying to understand your changes